code/modules/power/tesla/coil.dm

```diff
@@ -77,38 +87,45 @@
 		else
 			connect_to_network()
 
-/obj/machinery/power/tesla_coil/tesla_act(var/power)
+/obj/machinery/power/tesla_coil/zap_act(power, zap_flags)
 	if(anchored && !panel_open)
-		being_shocked = 1
 		//don't lose arc power when it's not connected to anything
 		//please place tesla coils all around the station to maximize effectiveness
-		var/power_produced = powernet ? power / power_loss : power
-		add_avail(power_produced*input_power_multiplier)
+		being_shocked = TRUE
+		addtimer(CALLBACK(src, .proc/reset_shocked), 1 SECONDS)
+		zap_buckle_check(power)
+		if(zap_flags & ZAP_GENERATES_POWER) //I don't want no tesla revolver making 8GW you hear
+			return power / 2
+		var/power_produced = powernet ? power * input_power_multiplier : power
+		add_avail(power_produced)
 		flick("coilhit", src)
-		playsound(src.loc, 'sound/magic/lightningshock.ogg', 100, 1, extrarange = 5)
-		tesla_zap(src, 5, power_produced)
-		addtimer(CALLBACK(src, .proc/reset_shocked), 10)
+		playsound(loc, 'sound/magic/lightningshock.ogg', 100, TRUE, extrarange = 5)
+		return power - power_produced //You get back the amount we didn't use
 	else
-		..()
+		. = ..()
 
 /obj/machinery/power/tesla_coil/proc/zap()
 	if((last_zap + zap_cooldown) > world.time || !powernet)
 		return FALSE
 	last_zap = world.time
-	var/coeff = (20 - ((input_power_multiplier - 1) * 3))
-	coeff = max(coeff, 10)
-	var/power = (powernet.avail/2)
+	var/power = (powernet.avail) * 0.2 * input_power_multiplier  //Always always always use more then you output for the love of god
+	power = min(surplus(), power) //Take the smaller of the two
 	add_load(power)
-	playsound(src.loc, 'sound/magic/lightningshock.ogg', 100, 1, extrarange = 5)
-	tesla_zap(src, 10, power/(coeff/2))
+	playsound(loc, 'sound/magic/lightningshock.ogg', 100, TRUE, extrarange = 5)
+	tesla_zap(src, 10, power, zap_flags)
+	zap_buckle_check(power)
 
 /obj/machinery/power/grounding_rod
 	name = "grounding rod"
-	desc = "Keep an area from being fried from Edison's Bane."
+	desc = "Keeps an area from being fried by Edison's Bane."
 	icon = 'icons/obj/tesla_engine/tesla_coil.dmi'
 	icon_state = "grounding_rod0"
-	anchored = 0
-	density = 1
+	anchored = FALSE
+	density = TRUE
+
+	can_buckle = TRUE
+	buckle_lying = 0
```

Comment:
Should this be TRUE/FALSE?

Reply:
This is not standardized well, unfortunately---this is actually not `TRUE/FALSE`; this is an angle that gets multiplied by this number.

Reply:
Oh..